packages/clerk-js/src/ui/components/UserVerification/UserVerificationFactorOnePassword.tsx

```diff
@@ -0,0 +1,95 @@
+import { useUser } from '@clerk/shared/react';
+import React from 'react';
+
+import { Col, descriptors, Flow, localizationKeys } from '../../customizables';
+import { Card, Form, Header, useCardState } from '../../elements';
+import { handleError, useFormControl } from '../../utils';
+import { HavingTrouble } from '../SignIn/HavingTrouble';
+import { useAfterVerification } from './use-after-verification';
+
+type UserVerificationFactorOnePasswordProps = {
+  onShowAlternativeMethodsClick: React.MouseEventHandler | undefined;
+};
+
+export function UserVerificationFactorOnePasswordCard(props: UserVerificationFactorOnePasswordProps): JSX.Element {
+  const { onShowAlternativeMethodsClick } = props;
+  const { user } = useUser();
+
+  const { handleVerificationResponse } = useAfterVerification();
+  const card = useCardState();
+
+  const [showHavingTrouble, setShowHavingTrouble] = React.useState(false);
+  const toggleHavingTrouble = React.useCallback(() => setShowHavingTrouble(s => !s), [setShowHavingTrouble]);
+
+  const passwordControl = useFormControl('password', '', {
+    type: 'password',
+    label: localizationKeys('formFieldLabel__password'),
+    placeholder: localizationKeys('formFieldInputPlaceholder__password'),
+  });
+
+  const handlePasswordSubmit: React.FormEventHandler = async e => {
+    e.preventDefault();
+    return user
+      ?.__experimental_verifySessionAttemptFirstFactor({
+        strategy: 'password',
+        password: passwordControl.value,
+      })
+      .then(handleVerificationResponse)
+      .catch(err => handleError(err, [passwordControl], card.setError));
+  };
+
+  if (showHavingTrouble) {
+    return <HavingTrouble onBackLinkClick={toggleHavingTrouble} />;
+  }
+
+  return (
+    <Flow.Part part='password'>
+      <Card.Root>
+        <Card.Content>
+          <Header.Root showLogo>
+            <Header.Title localizationKey={localizationKeys('__experimental_userVerification.password.title')} />
+            <Header.Subtitle localizationKey={localizationKeys('__experimental_userVerification.password.subtitle')} />
+          </Header.Root>
+          <Card.Alert>{card.error}</Card.Alert>
+          <Col
+            elementDescriptor={descriptors.main}
+            gap={4}
+          >
+            <Form.Root
+              onSubmit={handlePasswordSubmit}
+              gap={8}
+            >
+              {/* For password managers */}
+              {/*<input*/}
+              {/*  readOnly*/}
+              {/*  id='identifier-field'*/}
+              {/*  name='identifier'*/}
+              {/*  value={signIn.identifier || ''}*/}
+              {/*  style={{ display: 'none' }}*/}
+              {/*/>*/}
```

Comment:
@chanioxaris any ideas what we can use to solve this while the user is signed in ?

Reply:
We can use the existing function that picks up the identifier based on the user attributes